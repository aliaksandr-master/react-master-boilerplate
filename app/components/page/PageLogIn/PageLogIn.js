import { compose, withHandlers, withProps } from 'recompose';
import { connect } from 'react-redux-self';
import update, { callWhenAreInitialized } from 'react-updating-hoc';
import { referring } from 'react-composite-router';
import { checkAuthActionCreator } from '../../../reducers/user';
import instead from '../../hoc/instead';
import { RESPONSE_STATE } from '../../../config/constants';
import PageLogInComponent from './PageLogIn.view';
import reducer, { loginActionCreator } from './PageLogIn.store';



export default compose(
  connect({
    reducer,
    getters: [
      (self) => self,
      (_1, { user }) => user
    ],
    selector: ({ login }, user) => ({
      login,
      authState: user.authState,
      hasAuth: user.authenticated
    }),
    mapDispatchToProps: {
      loginAction: loginActionCreator,
      checkAuthAction: checkAuthActionCreator
    }
  }),
  withProps(() => ({ state: 'app' })),
  referring(),
  update(
    callWhenAreInitialized(({ checkAuthAction, hasAuth, state }) => {
      if (hasAuth) {
        state.apply();
        return;
      }

      checkAuthAction()
        .then(() => state.apply());
    })
  ),
  instead(({ authState }) => authState === RESPONSE_STATE.PENDING, () => null),
  withHandlers({
    handleSubmit: ({ state, loginAction }) => (values) => {
      loginAction(values)
        .then(() => state.apply());
    }
  })
)(PageLogInComponent);
