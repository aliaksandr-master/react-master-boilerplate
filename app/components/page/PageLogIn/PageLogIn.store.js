import { Dux } from '../../../lib/reducer';
import login from '../../../resources/api/login/post';
import commonRequestItemActionReducer from '../../../reducers/common-request-item-action';



const dux = Dux('PageLogIn');


// ACTION TYPES
const LOGIN = dux.actionType('LOGIN');


// ACTION CREATORS
export const loginActionCreator = dux.actionCreator(LOGIN, ({ email, password }) => ({
  apiAction: login({ email, password })
}));


// REDUCER
dux.reducer('login', commonRequestItemActionReducer('PageLogIn:login', LOGIN));

export default dux.compileReducer();
