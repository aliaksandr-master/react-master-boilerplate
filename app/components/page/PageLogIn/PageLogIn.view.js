import React from 'react';
import PropTypes from 'prop-types';
// import { Link } from 'react-composite-router';
import FractalField from 'react-fractal-field';
import Paper from 'material-ui/Paper';
import RaisedButton from 'material-ui/RaisedButton';
// import FlatButton from 'material-ui/FlatButton';
import injectSheet from 'react-jss';
import required from '../../../validate/required';
import maxLength from '../../../validate/max-length';
import { RESPONSE_STATE } from '../../../config/constants';
import email from '../../../validate/email';
import FieldInput, { SIZE as INPUT_SIZE } from '../../field/FieldInput';
import ErrorsListNotice from '../../ui/ErrorsListNotice';



const validateEmail = [ required(), maxLength(255), email() ];
const validatePassword = [ required(), maxLength(255) ];


const styles = () => ({
  root: {
    background: 'white',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
    height: '100%',
    minHeight: '100%',
    padding: 20
  },
  paper: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    padding: 30,
    width: '100%',
    maxWidth: 300
  },
  loginBtn: {
  },
  inputWr: {
    padding: '0 0 20px',
    width: 260
  },
  submitBtn: {
    float: 'right',
    color: 'white'
  },
  btnZone: {
    padding: '0'
  },
  h1: {
    fontSize: 21,
    lineHeight: 1,
    paddingBottom: 20
  }
});


const PageLogIn = ({ classes, submitting, handleSubmit, errorMessages }) => {
  return (
    <div className={classes.root}>
      <Paper rounded transitionEnabled className={classes.paper} zDepth={1}>
        <FractalField initialValue={{}} onSubmit={handleSubmit}>
          {({ triggerSubmit }) => (
            <div>
              <div className={classes.h1}>Tests</div>
              <div className={classes.inputWr}>
                <FractalField
                  name="email"
                  validate={validateEmail}
                >
                  {({ control, error, submitted }) => (
                    <FieldInput
                      {...control}
                      type="email"
                      name="email"
                      errorText={submitted ? error : ''}
                      placeholder="Email"
                      onEnterPress={triggerSubmit}
                      size={INPUT_SIZE.LARGE}
                    />
                  )}
                </FractalField>
              </div>

              <div className={classes.inputWr}>
                <FractalField
                  name="password"
                  validate={validatePassword}
                >
                  {({ control, error, submitted }) => (
                    <FieldInput
                      {...control}
                      name="password"
                      errorText={submitted ? error : ''}
                      type="password"
                      placeholder="Password"
                      onEnterPress={triggerSubmit}
                      size={INPUT_SIZE.LARGE}
                    />
                  )}
                </FractalField>
              </div>

              <ErrorsListNotice
                messages={errorMessages}
              />
              <div className={classes.btnZone}>
                {/*<FlatButton
                  className={classes.loginBtn}
                  onClick={triggerSubmit}
                  disabled={submitting}
                  containerElement={<Link state="register" />}
                >
                  Register
                </FlatButton>*/}

                <RaisedButton
                  primary
                  className={classes.submitBtn}
                  onClick={triggerSubmit}
                  disabled={submitting}
                  type="submit"
                >
                  Sign In
                </RaisedButton>
              </div>
            </div>
          )}
        </FractalField>
      </Paper>
    </div>
  );
};



PageLogIn.propTypes = {
  classes: PropTypes.object,
  login: PropTypes.shape({
    responseState: PropTypes.oneOf(Object.values(RESPONSE_STATE))
  }),
  handleSubmit: PropTypes.func, // from redux-form
  submitting: PropTypes.bool,
  errorMessages: PropTypes.arrayOf(PropTypes.string)
};



PageLogIn.defaultProps = {
};



export default injectSheet(styles)(PageLogIn);
