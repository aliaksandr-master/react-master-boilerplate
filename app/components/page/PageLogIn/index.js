import PageLogIn from './PageLogIn';



export default PageLogIn;
