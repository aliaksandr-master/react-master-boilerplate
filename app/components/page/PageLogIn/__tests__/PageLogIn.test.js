/* eslint-env browser, jest */


import React from 'react';
import renderer from 'react-test-renderer';
import PageLogIn from '../PageLogIn';
import Test from '../../../test/Test';



describe('rendering', () => {
  it('should render something', () => {
    const component = renderer.create(
      <Test>
        <PageLogIn redirectToRoot={false} />
      </Test>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
