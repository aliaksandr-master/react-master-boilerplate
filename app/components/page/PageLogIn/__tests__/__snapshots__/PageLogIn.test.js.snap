// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render something 1`] = `null`;
